tool.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+function load_config_file {
+	local dummycfg_path=/src/dummy.cfg
+
+	if [ -f $dummycfg_path ]; then
+		codacy_foobar_parameter="$(tr -d '\n' < "$dummycfg_path")"
+	fi
+	codacy_patterns="foobar"
+}
+
+function load_src_files {
+	codacy_files=$(cd /src || exit; find . -type f -exec echo {} \; | cut -c3-)
+}
+
+if [ -f /.codacyrc ]; then
+	#parse
+	codacyrc_file=$(jq -erM '.' < /.codacyrc)
+
+	# error on invalid json
+	if [ $? -ne 0 ]; then
+		echo "Can't parse .codacyrc file"
+		exit 1
+	fi
+
+	codacy_files=$(jq -cer '.files | .[]' <<< "$codacyrc_file")
+	codacy_patterns="$(jq -cer ".tools | .[] | select(.name==\"dummy\") | .patterns | .[].patternId" <<< "$codacyrc_file")"
+	codacy_foobar_parameter="$(jq -cer ".tools | .[] | select(.name==\"dummy\") | .patterns | .[].parameters | .[] | select(.name==\"value\") | .value" <<< "$codacyrc_file")"
+
+	if [ $? -ne 0 ]; then
+		unset codacy_foobar_parameter
+	fi
+
+	if [ "$codacy_patterns" == "" ]; then
+		# When no supplied patterns
+		load_config_file
+	fi
+
+	if [ "$codacy_files" == "" ]; then
+		# When no files given, run with all files in /src
+		load_src_files
+	fi
+else
+	load_config_file
```

Comment:
call these functions just once with an or in the condition removing this `else` branch